### ISSUE TITLE
Allow short ternary operator for boolean

### ISSUE THREAD

**Issue by LastDragon-ru** [phpstan/phpstan-strict-rules#268]
> Disallow short ternary operator (?:) - implies **weak comparison**, it's recommended to use null coalesce operator (??) or ternary operator with strict condition.

It is not the case if the left value is always boolean, right?

```php

$a = false;      // boolean
$b = $a ?: null; // much more readable than
$c = $a !== false ? true : null;
```

**Comment by VincentLanglet** [comment]
You can just write 
```
$c = $a ? true : null;
```

**Comment by LastDragon-ru** [comment]
It is still too long)

**Comment by ondrejmirtes** [comment]
Rejected, reasons described here https://github.com/phpstan/phpstan-strict-rules/pull/287#issuecomment-3370490453
